Source/mpqapi.cpp

```diff
@@ -581,21 +581,25 @@ void mpqapi_store_modified_time(const char *pszArchive, int dwChar)
 	}
 }
 
-void mpqapi_flush_and_close(const char *pszArchive, BOOL bFree, int dwChar)
+BOOL mpqapi_flush_and_close(const char *pszArchive, BOOL bFree, int dwChar)
 {
-	if (sghArchive != INVALID_HANDLE_VALUE) {
-		if (save_archive_modified) {
-			if (mpqapi_can_seek()) {
-				if (WriteMPQHeader()) {
-					if (mpqapi_write_block_table())
-						mpqapi_write_hash_table();
-				}
-			}
+    BOOL ret = FALSE;
+    if (sghArchive == INVALID_HANDLE_VALUE)
+		ret = TRUE;
+	else {
+        ret = FALSE;
+        if (!save_archive_modified)

```

Comment:
Indentation appears to be mixed spaces and tabs, but we will fix it once we format the file when it's bin exact :)